### DISCUSSION TITLE
"unable to locate ObjectStore plugin named velero.io/azure"

### DISCUSSION THREAD

**Discussion by arthursantana** [vmware-tanzu/velero#3333] [Community support Q&A] [answered]
**What steps did you take and what happened:**

Velero was installed with:
```bash
velero install \
    --provider azure \
    --plugins velero/velero-plugin-for-microsoft-azure:v1.1.0 \
    --bucket $BLOB_CONTAINER \
    --secret-file ./credentials-velero \
    --backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP, storageAccount=$AZURE_STORAGE_ACCOUNT_ID \
    --snapshot-location-config apiTimeout=5m,resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP \
    --use-restic --wait
```

```bash
$ velero server log
[... edited ...]
An error occurred: some backup storage locations are invalid: error getting backup store for location "default": unable to locate ObjectStore plugin named velero.io/azure
```

Trying to create backups afterwards results in a failure with no errors or warnings.

**What did you expect to happen:**

Velero should start normally and backups should succeed or show a reason for failure.

**The output of the following commands will help us better understand what's going on**:

```bash
$ velero backup describe b1
Name:         b1
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.19.3
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=19

Phase:  Failed (run `velero backup logs b1` for more information)

Errors:    0
Warnings:  0

Namespaces:
  Included:  *
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

Hooks:  <none>

Backup Format Version:  1

Started:    2021-01-19 14:45:31 -0300 -03
Completed:  2021-01-19 14:46:21 -0300 -03

Expiration:  2021-02-18 14:45:31 -0300 -03

Total items to be backed up:  463
Items backed up:              463

Velero-Native Snapshots:  4 of 4 snapshots completed successfully (specify --details for more information)
```
- `velero backup logs <backupname>`
https://gist.github.com/arthursantana/75579da7aee2b8861bc351b1145918a2

- `kubectl logs deployment/velero -n velero`
https://gist.github.com/arthursantana/8bb413798d842e3c25b159566530d799

- `velero backup logs <backupname>`
https://gist.github.com/arthursantana/384d33e354cdc2ba325d195ea87cc009

**Anything else you would like to add:**
My problem is similar to https://github.com/vmware-tanzu/velero/issues/2303#issue-571919887 , but I couldn't find a solution in this thread.

Output for the relevant commands requested earlier in the thread:
```bash
$ kubectl -n velero get deploy velero -ojson | jq .spec.template.spec.initContainers
[
  {
    "image": "velero/velero-plugin-for-microsoft-azure:v1.1.0",
    "imagePullPolicy": "IfNotPresent",
    "name": "velero-plugin-for-microsoft-azure",
    "resources": {},
    "terminationMessagePath": "/dev/termination-log",
    "terminationMessagePolicy": "File",
    "volumeMounts": [
      {
        "mountPath": "/target",
        "name": "plugins"
      }
    ]
  }
]
```

```bash
nobody@velero-dc8c47b79-gfwd9:/$ ls plugins/
velero-plugin-for-microsoft-azure
```

**Environment:**

- Velero version (use `velero version`): 
```
$ velero version
Client:
	Version: v1.4.2
	Git commit: 56a08a4d695d893f0863f697c2f926e27d70c0c5
Server:
	Version: v1.4.2
```

- Velero features (use `velero client config get features`): 
```features: <NOT SET>```

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"archive", BuildDate:"2020-11-25T13:19:56Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-16T20:43:08Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration:
Azure Kubernetes Service

- OS (e.g. from `/etc/os-release`):
Manjaro Linux

**Vote on this issue!**
This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

```

**Comment by zubron** [comment]
Hi @arthursantana! In order to get the logs for the Velero deployment, you will need to look at the logs from the pod in your cluster which you have already provided with [`kubectl logs deployment/velero -n velero`](https://gist.github.com/arthursantana/8bb413798d842e3c25b159566530d799). The `velero server` command is the command used in the Velero pod to run the server and is not for interacting with an existing installation. 

Looking at the logs, this line stands out: https://gist.github.com/arthursantana/8bb413798d842e3c25b159566530d799#file-gistfile1-txt-L102
It may suggest that your storage location is not accessible from the cluster? You can verify whether your storage locations are ready to use by checking the output from `velero backup-location get`. Your storage location should have the phase "Available". That indicates that Velero can connect to it successfully.

**Comment by arthursantana** [comment]
Hi @zubron! Thank you for your answer.

Here's the output of `velero backup-location get`:
```
NAME      PROVIDER   BUCKET/PREFIX   ACCESS MODE
default   azure      velero-extra    ReadWrite
```

**Comment by dsu-igeek** [comment, marked as answer]
We're getting this error message at the end of your backup
`time="2021-01-19T17:46:21Z" level=error msg="backup failed" controller=backup error="rpc error: code = Unavailable desc = transport is closing" key=velero/b1 logSource="pkg/controller/backup_controller.go:273"
`

We've seen an issue with the Azure plug-in that it needs more memory than was being allocated by default for the plug-in - https://github.com/vmware-tanzu/velero/issues/3234

Please check and see if the memory limit on your Velero pod is at least 512M.  If not, please increase it to 512M and try it again or install Velero 1.5.3 (https://github.com/vmware-tanzu/velero/releases/tag/v1.5.3) which should patch the limit.

**Comment by arthursantana** [comment]
Hi @dsu-igeek , updating to 1.5.3 seems to have worked perfectly. Thank you very much.

**Comment by miechv** [comment]
Hi, i'm struggling with the same error but unfortunately even tough i'm using the newest version of the helm chart with 512M included. In my values.yaml i'm using the velero/velero-plugin-for-microsoft-azure:v1.6.0 image and velero.io/azure provider for both backupstoragelocation and volumesnapshotlocation.



`ERRO[0001] Error getting backup store for this location  backupLocation=velero/default controller=backup-sync error="unable to locate ObjectStore plugin named velero.io/azure" logSource="pkg/controller/backup_sync_controller.go:100"`

Basically this is the only error log that i'm receiving after trying to create a backup.

I checked the available plugins and the velero.io/azure with ObjectStore kind is present.

`velero get plugin | grep azure                                                                                        
velero.io/azure                             ObjectStore
velero.io/azure                             VolumeSnapshotter
`

Is there anybody who can help me with this?
